src/Commands/ApigeeEdgeCommands.php

```diff
@@ -57,4 +71,61 @@ public function sync() {
     $this->cliService->sync($this->io(), 'dt');
   }
 
+  /**
+   * Create a role in an Apigee organization.
+   *
+   * Create role with proper permissions for connecting a Drupal
+   * Apigee Edge module to the Edge org. You must run this script
+   * as a user with the orgadmin role.
+   *
+   * @param string $org
+   *   The Apigee Edge org to create the role in.
+   * @param string $email
+   *   An Apigee user email address with orgadmin role for this org.
+   * @param array $options
+   *   Drush options for the command.
+   *
+   * @option password
+   *   Password for the Apigee orgadmin user. If not set, you will be prompted
+   *   for the password.
+   * @option base-url
+   *   Base URL to use, defaults to public cloud URL.
+   * @option role-name
+   *   The role to create in the Apigee Edge org.
+   * @usage drush create-edge-role myorg me@example.com
+   *   Create "drupalportal" role as orgadmin me@example.com for org myorg.
+   * @usage drush create-edge-role myorg me@example.com --base-url=https://api.edge.example.com
+   *   Create role on private Apigee Edge server "api.edge.example.com".
+   * @usage drush create-edge-role myorg me@example.com --role-name=portal
+   *   Create role named "portal".
+   * @command apigee-edge:create-edge-role
+   * @aliases create-edge-role
+   */
+  public function createEdgeRole(
+    string $org,
+    string $email,
+    array $options = [
+      'password' => NULL,
+      'base-url' => self::DEFAULT_BASE_URL,
+      'role-name' => self::DEFAULT_ROLE_NAME,
+    ]) {
+    $this->edgeConnectionUtilService->createEdgeRoleForDrupal($this->io(), 'dt', $org, $email, $options['password'], $options['base-url'], $options['role-name']);
```

Comment:
User input should be validated, ex.: base-url should be a valid URI, role name should be string, etc. Symfony's options resolver could handle this easily.

Reply:
Turns out that role can be a number and I am not sure what the rules are, so we can let the Apigee API respond back with any errors. Added validation for Base URL however.